src/lib/Timeline.js

```diff
@@ -387,14 +395,15 @@ export default class ReactCalendarTimeline extends Component {
     // if the items or groups have changed we must re-render
     const forceUpdate = items !== prevState.items || groups !== prevState.groups
 
-    // We are a controlled component
-    if (visibleTimeStart && visibleTimeEnd) {
+    // We want to trigger a refresh with updated time range based on zoom
+    if (remoteZoomTimeStart !== prevState.remoteZoomTimeStart || remoteZoomTimeEnd !== prevState.remoteZoomTimeEnd) {
```

Comment:
was this enough to limit it from always firing? or is it better because you arent messing with visibleTimeStart/visibleTimeStart in the derivedState.  So its not firing a lot less.

Reply:
Yes, it only fires now when we update this new prop (triggered by the zoom, or by changes to the groups/items) so that we can be more explicit when we want to trigger a change... it's also only triggering if there is a difference obviously between when we last updated it in state vs the most recent prop.  Basically the only time a re-render then triggers is with changes to stops (items) or groups (routes), or when you use the separate / external zoom control.

Reply:
Awesome, is there a version of this pushed yet to test with?  I think you mentioned a debug tag?

Reply:
Yes, sent you details.  I'm wondering now though whether it may make sense to leave in the visibleTimeStart/End, to allow it to remain fully controlled in case that's how others might want to use it (and thus backward compatible).